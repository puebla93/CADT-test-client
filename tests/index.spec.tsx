import { render, screen, fireEvent } from '@testing-library/react'
import Home from '../pages/index'
import '@testing-library/jest-dom'

describe('Home page test', () => {
    test('Home matches snapshot', async () => {
        const { container } = render(<Home />)
        expect(container).toMatchSnapshot()
    });

    test('Home page have images components', () => {
        render(<Home />)

        const images = screen.getAllByRole('img');

        expect(images).toHaveLength(2);
        expect(images[0]).toHaveAttribute('src', '/placeholder.png');
        expect(images[0]).toHaveAttribute('alt', '');
        expect(images[1]).toHaveAttribute('src', '/placeholder.png');
        expect(images[1]).toHaveAttribute('alt', '');
    });

    test('Home page have buttons components', () => {
        render(<Home />)

        const button = screen.getByRole('button');

        expect(button).toBeInTheDocument();
        expect(button).toBeDisabled();
        expect(button).toHaveAttribute('type', 'submit');
    });

    test('Home page have select components', () => {
        render(<Home />)

        const optionEmboss = screen.getByText('Emboss');
        expect(optionEmboss).toBeInTheDocument();
        expect(optionEmboss).toHaveAttribute('value', 'emboss');

        const optionGaussianBlur = screen.getByText('Gaussian Blur');
        expect(optionGaussianBlur).toBeInTheDocument();
        expect(optionGaussianBlur).toHaveAttribute('value', 'gaussianBlur');

        const optionCannyEdge = screen.getByText('Canny Edge');
        expect(optionCannyEdge).toBeInTheDocument();
        expect(optionCannyEdge).toHaveAttribute('value', 'cannyEdge');

        const optionGrayScale = screen.getByText('Gray Scale');
        expect(optionGrayScale).toBeInTheDocument();
        expect(optionGrayScale).toHaveAttribute('value', 'grayScale');
        
        const optionSepia = screen.getByText('Sepia');
        expect(optionSepia).toBeInTheDocument();
        expect(optionSepia).toHaveAttribute('value', 'sepia');
    });
})
