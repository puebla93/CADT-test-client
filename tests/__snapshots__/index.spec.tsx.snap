// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test Home matches snapshot 1`] = `
<div>
  <div
    class="sc-dkrFOg gTMfdM"
  >
    <div
      class="sc-gswNZR jXYaRP"
    >
      <img
        alt=""
        src="/placeholder.png"
      />
      <img
        alt=""
        src="/placeholder.png"
      />
    </div>
    <select
      class="sc-bcXHqe hwNYZh"
    >
      <option
        value="emboss"
      >
        Emboss
      </option>
      <option
        value="gaussianBlur"
      >
        Gaussian Blur
      </option>
      <option
        value="cannyEdge"
      >
        Canny Edge
      </option>
      <option
        value="grayScale"
      >
        Gray Scale
      </option>
      <option
        value="sepia"
      >
        Sepia
      </option>
    </select>
    <div
      class="sc-hLBbgP beUmJm"
    >
      <label
        class="sc-jSUZER gbvLtW"
      >
        <input
          accept="image/png, image/jpeg"
          type="file"
        />
        Upload image
      </label>
      <button
        class="sc-eDvSVe jMKZCv"
        disabled=""
        type="submit"
      >
        Process image
      </button>
    </div>
  </div>
</div>
`;
